src/server/verification/processor/EnrollmentSession.js

```diff
@@ -49,32 +64,49 @@ export default class EnrollmentSession {
     return result
   }
 
+  initialize(payload: any) {
+    const { gun } = this
+    const { sessionId } = payload
+
+    this.sessionRef = gun.session(sessionId)
+    // returning this to allow initialize &
+    // get sessionRef via destructuring in a single call
+    return this
+  }
+
   onEnrollmentStarted() {
     const { sessionRef } = this
 
     sessionRef.put({ isStarted: true })
   }
 
   onEnrollmentProcessing(processingPayload: IEnrollmentEventPayload) {
-    const { sessionRef } = this
+    const { sessionRef, log } = this
+
+    if ('isDuplicate' in processingPayload) {
+      log.info('Checking for duplicates:', processingPayload)
+    }
+
+    if ('isEnrolled' in processingPayload) {
+      log.info('Checking for liveness and tried to enroll:', processingPayload)
+    }
 
     sessionRef.put(processingPayload)
   }
 
   async onEnrollmentCompleted() {
-    const { sessionRef, user, storage, adminApi } = this
+    const { sessionRef, user, storage, adminApi, log } = this
     const { gdAddress, profilePublickey, loggedInAs } = user
 
-    try {
-      await Promise.all([
-        adminApi.whitelistUser(gdAddress, profilePublickey),
-        storage.updateUser({ identifier: loggedInAs, isVerified: true })
-      ])
-
-      sessionRef.put({ isWhitelisted: true })
-    } catch ({ message }) {
-      sessionRef.put({ isWhitelisted: false, isError: message })
-    }
+    log.info('Whitelistening user:', loggedInAs)
+
+    await Promise.all([
+      adminApi.whitelistUser(gdAddress, profilePublickey),
```

Comment:
verify that user is not whitelisted before calling this. if he is whitelisted and we call this there will be an exception
i think its adminApi.isCitizen

Reply:
it doesnt needed as whitelistUser already perform this check:
```javascript
async whitelistUser(address: string, did: string): Promise<TransactionReceipt | boolean> {
const isVerified = await this.isVerified(address)
    if (isVerified) {
      return { status: true }
    }
```